Cython/Compiler/ExprNodes.py

```diff
@@ -3672,8 +3676,8 @@ def analyse_target_declaration(self, env):
 
     def analyse_as_type(self, env):
         base_type = self.base.analyse_as_type(env)
-        if base_type and (not base_type.is_pyobject or base_type.is_python_type_constructor):
-            if base_type.is_cpp_class or base_type.is_python_type_constructor:
+        if base_type and (not base_type.is_pyobject or base_type.python_type_constructor_name):
+            if base_type.is_cpp_class or base_type.python_type_constructor_name:
                 if isinstance(self.index, TupleNode):
```

Comment:
I think it might be enough to ask for `is_sequence_constructor` instead of the type check.
```suggestion
                if self.index.is_sequence_constructor:
```